FindPythonDev.cmake

```diff
@@ -57,11 +57,11 @@ if (PYTHON_EXECUTABLE)
 else ()
     find_program(PYTHON_CONFIG NAMES
                  python3-config
+                 python3.11-config
+                 python3.10-config
```

Comment:
Just a thought in passing, we remove the older unsupported versions from this list so users who have just say `python3.5-config` might see an error about no Python being found instead of a missed version requirement which seems not ideal. 

Looking over this file, I wonder whether it is actually still needed now that python2 support is dropped (even though we still check for python2-config [above](https://github.com/zeek/cmake/blob/master/FindPythonDev.cmake#L48-L55)). Without checking, it would seem that the [`FindPython.cmake`](https://cmake.org/cmake/help/latest/module/FindPython.html) which ships with any of the versions of CMake we require would be sufficient with the following replacements:

- `PYTHON_EXECUTABLE` -> `Python_EXECUTABLE`
- `PYTHON_LIBRARIES` -> `Python_LIBRARIES`
- `PYTHON_INCLUDE_DIR` -> `PYTHON_INCLUDE_DIRS`
- `PYTHON_CONFIG` has no equivalent, but seems unused outside of this file.

Could we use the following much simpler code which should also give better diagnostics?

```cmake
find_package(Python 3.7 REQUIRED COMPONENTS Development Interpreter)
```

Reply:
There's a few other files that we vendor out of CMake as well: FindOpenSSL.cmake, FindPackageHandleStandardArgs.cmake, and SelectLibraryConfigurations.cmake. I haven't spent any time looking at whether we can remove those though, or if there's any significant differences between ours and the upstream ones.

Reply:
I'm all for removing these if we can. It's hard to identify from the revision history why we added `FindPythonDev.cmake` , or whether we added anything critical over time. The good news is that it's only used in Broker and pysubnettree, so it's easy to try out. There's also some use of `find_package(PythonInterp)` (in trace-summary and again Broker), which the upstream `FindPython.cmake` seems to support as well, so we could modernize all of this in one go.

@Neverlord I think these weren't on your immediate TODO list because you're focusing on the plugin features at the moment, right?